test/test_cuda.py

```diff
@@ -3259,6 +3260,16 @@ def test_graph_capture_simple(self):
 
         self.assertTrue(b.sum().item() == 11000.)
 
+    @unittest.skipIf((not TEST_CUDA) or
+                     TEST_WITH_ROCM or
+                     int(torch.version.cuda.split(".")[0]) < 11, "CUDA >= 11.0 required for graphs")
+    def test_graph_warn_if_has_zero_nodes(self):
+        with warnings.catch_warnings(record=True) as caught:
+            g = torch.cuda.CUDAGraph()
+            g.capture_begin()
+            g.capture_end()
+        self.assertTrue(any("The CUDA Graph is empty" in str(w.message) for w in caught))
```

Comment:
setting stream is missing here, but when added there's a weird interaction of streams and warnings. See:
```py
In [1]: import warnings

In [2]: import torch

In [3]: a = torch.randn(3, 3, device="cuda")

In [4]: out = torch.empty_like(a)

In [5]: with warnings.catch_warnings(record=True) as w:
   ...:     torch.linalg.cond(a, 2, out=out)
   ...:     print(w)
   ...:
[<warnings.WarningMessage object at 0x7ffbe05dce50>]

In [6]: with warnings.catch_warnings(record=True) as w:
   ...:     s = torch.cuda.Stream()
   ...:     with torch.cuda.stream(s):
   ...:         torch.linalg.cond(a, 2, out=out)
   ...:     print(w)
   ...:
[]
```

Reply:
Example was incorrect.